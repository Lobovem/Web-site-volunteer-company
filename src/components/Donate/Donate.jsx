import { Btn } from '../kit/Btn/Btn';
import s from './Donate.module.scss';

export const Donate = () => {
  return (
    <div className={s.donate}>
      <div className={s.donate__wrap}>
        <div className={s.donate__header}>
          <p className={`${s.donate__headerTitle} ${s.donate__headerTitle_first}`}>підтримай україну</p>
          <p className={`${s.donate__headerTitle} ${s.donate__headerTitle_second}`}>підтримай україну</p>
          <p className={`${s.donate__headerTitle} ${s.donate__headerTitle_third}`}>підтримай україну</p>
        </div>

        <div className={s.donate__main}>
          <h2 className={s.donate__mainTitle}>Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!</h2>
          <p className={s.donate__mainDesc}>Кожна гривня для нас дуже важлива, тому зробити свій внесок ви можете саме зараз</p>

          <Btn title="задонатити" className="btn"></Btn>
        </div>
      </div>
    </div>
  );
};
