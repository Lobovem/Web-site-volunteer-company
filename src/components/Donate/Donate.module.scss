@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.donate {
  background: $color-grey;

  &__wrap {
    @include layoutMain;
    padding: 50px 0;

    @include sm-min {
      max-width: 640px;
    }

    @include lg-min {
      max-width: 100%;
      padding: 120px 0;
    }
  }

  &__header {
    background: $gradient-main;
    box-shadow: 0px 4px 15px rgba(2, 95, 167, 0.08);
    border-radius: 15px 15px 0px 0px;
    min-height: 100px;
    padding: 12px 20px;

    @include lg-min {
      padding: 0px;
      gap: 40px;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }

  &__headerTitle {
    font-weight: $font-weight-500;
    font-size: $font-size-20;
    line-height: 20px;
    text-transform: uppercase;
    color: $color-grey;

    mix-blend-mode: normal;
    opacity: 0.2;
    margin: 10px 0 10px 20px;

    &_second {
      text-align: center;
    }

    &_third {
      text-align: right;

      @include lg-min {
        text-align: center;
      }
    }

    @include lg-min {
      display: inline-block;
      font-size: $font-size-24;
      line-height: 36px;
      margin: 0;
    }

    @media (min-width:1210px) {
      font-size: 32px;
      line-height: 36px;
    }
  }

  &__main {
    background: $color-white;
    padding: 50px 26px 70px 26px;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 34px;
    box-shadow: 0px 4px 15px rgba(2, 95, 167, 0.08);
    border-radius: 0px 0px 15px 15px;

    @include sm-min {
      padding: 90px 100px;
      gap: 40px;

    }

    @include lg-min {
      padding: 160px 0;
      gap: 50px;
    }
  }

  &__mainTitle {
    font-weight: $font-weight-500;
    font-size: $font-size-30;
    line-height: 36px;
    text-align: center;
    color: $color-black;
    mix-blend-mode: normal;
    background: $gradient-main;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    margin: 0;
    max-width: 220px;

    @include xs-min {
      max-width: 300px;
    }

    @include lg-min {
      font-size: $font-size-48;
      line-height: 57px;
      max-width: 700px;
    }
  }

  &__mainDesc {
    font-weight: $font-weight-500;
    font-size: $font-size-20;
    line-height: 24px;
    text-align: center;
    color: $color-black;
    mix-blend-mode: normal;
    margin: 0;

    @include lg-min {
      font-size: $font-size-24;
      max-width: 770px;
    }
  }
}