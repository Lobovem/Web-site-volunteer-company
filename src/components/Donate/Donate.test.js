import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Donate } from './Donate';

describe('Donate component', () => {
  test('renders Donate component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Router>
        <Donate />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  test('search title tag h2', () => {
    render(
      <Router>
        <Donate />
      </Router>
    );
    const h1Element = screen.getByRole('heading', { level: 2 });
    expect(h1Element).toBeInTheDocument();
  });

  test('renders and search component btn', () => {
    render(
      <Router>
        <Donate />
      </Router>
    );
    const buttonElement = screen.getByRole('button', { type: 'button' });
    expect(buttonElement).toBeInTheDocument();
  });
});
