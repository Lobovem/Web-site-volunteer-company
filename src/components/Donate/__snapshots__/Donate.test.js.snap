// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donate component renders Donate component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="donate"
      >
        <div
          class="donate__header"
        >
          <p
            class="donate__headerTitle donate__headerTitle_first"
          >
            підтримай україну
          </p>
          <p
            class="donate__headerTitle donate__headerTitle_second"
          >
            підтримай україну
          </p>
          <p
            class="donate__headerTitle donate__headerTitle_third"
          >
            підтримай україну
          </p>
        </div>
        <div
          class="donate__main"
        >
          <h2
            class="donate__mainTitle"
          >
            Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!
          </h2>
          <p
            class="donate__mainDesc"
          >
            Кожна гривня для нас дуже важлива, тому зробити свій внесок ви можете саме зараз
          </p>
          <a
            href="/donate"
          >
            <button
              class="btn"
              type="button"
            >
              задонатити
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="donate"
    >
      <div
        class="donate__header"
      >
        <p
          class="donate__headerTitle donate__headerTitle_first"
        >
          підтримай україну
        </p>
        <p
          class="donate__headerTitle donate__headerTitle_second"
        >
          підтримай україну
        </p>
        <p
          class="donate__headerTitle donate__headerTitle_third"
        >
          підтримай україну
        </p>
      </div>
      <div
        class="donate__main"
      >
        <h2
          class="donate__mainTitle"
        >
          Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!
        </h2>
        <p
          class="donate__mainDesc"
        >
          Кожна гривня для нас дуже важлива, тому зробити свій внесок ви можете саме зараз
        </p>
        <a
          href="/donate"
        >
          <button
            class="btn"
            type="button"
          >
            задонатити
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
